### ISSUE TITLE
PC-98 Amaranth MIDI support does not work

### ISSUE THREAD

**Issue by joncampbell123** [joncampbell123/dosbox-x#937]
**Describe the bug**
Amaranth claims to support MIDI, but it does not work in DOSBox-X, and game cannot proceed past initial screens

**To Reproduce**
1. Boot Amaranth disk image
2. Select "GS" sound module (general MIDI)
3. Start game
4. Hear no music, game eventually gets stuck

**Comment by joncampbell123** [comment]
Several other games also support MIDI but fail to use it.

**Comment by yksoft1** [comment]
Super Shanghai: Dragon's Eye hangs "Sending MIDI instruments data".
![image](https://user-images.githubusercontent.com/37106774/49054850-9eb7d580-f230-11e8-90c7-fcaeedc8526b.png)

**Comment by joncampbell123** [comment]
If the debugger is available, use the "PIC" command to see which interrupts are masked, unmasked, request and in service. The game may have mis-detected the IRQ.

As of 0.82.11 MIDI defaults to IRQ 6.

Also use the "D" command to view the interrupt table at 0:20 (memory location of IRQ 8) and look for vectors that do NOT point into the BIOS (taken by the game).

**Comment by yksoft1** [comment]
> If the debugger is available, use the "PIC" command to see which interrupts are masked, unmasked, request and in service. The game may have mis-detected the IRQ.
> 
> As of 0.82.11 MIDI defaults to IRQ 6.
> 
> Also use the "D" command to view the interrupt table at 0:20 (memory location of IRQ 8) and look for vectors that do NOT point into the BIOS (taken by the game). It's possible the game mis-detected the IRQ and hooked IRQ 5, for example.

I don't think that it misdetected MPU-401 interrupt as there is such log recorded.
`  87537259       ALL:Sysex message size 11`

By the way, this game tried to write E0000-E7FFF for unknown reason.

**Comment by joncampbell123** [comment]
A lot of games seem to write E0000-E7FFF before switching on analog/16-color mode for some reason. E0000-E7FFF would be the 4th bitplane for 16-color mode if it was enabled.

**Comment by joncampbell123** [comment]
Sending one message to the MPU-401 does not mean it found the interrupt. The game might rely on the MIDI interrupt to send additional MIDI commands.

**Comment by yksoft1** [comment]
I don't have enough time to test iodelay and cycles settings thoroughly for Fuga's MIDI driver FMD.EXE. 
[FMD.zip](https://github.com/joncampbell123/dosbox-x/files/2721215/FMD.zip)

However I can see that MIDI code in Neko Project II (/cbus/mpu98ii.c) implemented something about waiting in MIDI port output handlers (mpu98ii_o0 and mpu98ii_o2).

**Comment by joncampbell123** [comment]
@yksoft1 I'll see what Neko Project II is doing for MPU-401 timing.

On a related note, what is the bus clock frequency of C-Bus?

On the original IBM PC, the ISA bus clock was synchronous with the CPU (4.77Mhz). Is the same true for the original PC-98 and the C-Bus?

**Comment by yksoft1** [comment]
Except the first generation, which C-Bus was synchronous with its 5MHz 8086, PC-98s before the age of SuperIO and PCI use either 10MHz (9.8304MHz) or 8MHz (7.9872MHz) for its C-Bus. 
It's determined by the CPU clock base (2.4756Mhz or 1.9968MHz). For example, on a 16MHz 386, C-Bus runs at 8MHz and on a 25MHz 386, C-Bus runs at 10MHz.
After NEC brought SuperIO and PCI to PC-98, C-Bus clock no longer ties to the CPU oscillator and got fixed to 10MHz.
